### PR TITLE
[HOPSWORKS-3054] Stream Feature Group Python Engine ingestion

### PR DESCRIPTION
**Pull request by moritzmeister**

### CODE REVIEW

**Review by SirOibaf — CHANGES_REQUESTED** [review]
Some high level questions:
- I thought we agreed on setting `stream=True` for all the feature group created using the Python engine. I don't see it but maybe I missed in a previous PR.
- The `insert_stream` Python method still throws a NotImplementedException. Should we leave it like this? I guess so.
- There is a `python/test.py` file hanging around at the bottom of the PR. Should we remove it?

File: python/hsfs/client/base.py [hunk 1]
```diff
@@ -91,6 +91,29 @@ def _read_file(self, secret_file):
...
+            f.write(content)
+
+    def _get_ca_chain_path(self) -> str:
+        return os.path.join("/tmp", "ca_chain.pem")
+
+    def _get_client_cert_path(self) -> str:
+        return os.path.join("/tmp", "client_cert.pem")
+
+    def _get_client_key_path(self) -> str:
+        return os.path.join("/tmp", "client_key.pem")
```
[the remark this answers is not in the document]
Comment (moritzmeister) [reply python/hsfs/client/base.py:115]: How can he "request" it? I thought we said we are moving away from the cert folder and always put it in `tmp`

File: python/hsfs/engine/python.py [hunk 2]
```diff
@@ -20,23 +20,40 @@
...
+from hsfs.feature_group import FeatureGroup
+
+HAS_FAST = False
+try:
+    from fastavro import schemaless_writer
+    from fastavro.schema import parse_schema
+
+    HAS_FAST = True
+except ImportError:
+    pass
```
[the remark this answers is not in the document]
Comment (moritzmeister) [reply python/hsfs/engine/python.py:56]: just if you uninstall it manually.

File: python/hsfs/engine/python.py [hunk 3]
```diff
@@ -454,3 +521,126 @@ def add_file(self, file):
...
+            )
+
+        # make sure producer blocks and everything is delivered
+        producer.flush()
+
+        # start backfilling job
+        job_name = "{fg_name}_{version}_offline_fg_backfill".format(
+            fg_name=feature_group.name, version=feature_group.version
+        )
+        job = self._job_api.get(job_name)
```
[the remark this answers is not in the document]
Comment (moritzmeister) [reply python/hsfs/engine/python.py:584]: I thought it's nice to return the job anyway, even when you are not starting it immediately, then later, they can use hopsworks-api to start it and they know the name of the job etc.

**Review by SirOibaf — APPROVED** [review]

**Pull request merged**